src/string.cr

```diff
@@ -2200,6 +2200,66 @@ class String
     match_result.try &.begin(0)
   end
 
+  # Searches sep or pattern (regexp) in the string from the end of the string,
+  # and returns the part before it, the match, and the part after it.
+  # If it is not found, returns two empty strings and str.
+  #
+  # ```
+  # "hello".rpartition("l")  # => ["hel", "l", "o"]
```

Comment:
The example shows that `rpartition` returns an array, but that is not the case. Please check that the code samples in the docs matches the real output.